### ISSUE TITLE
Learning rate scheduler

### ISSUE THREAD

**Issue by leemengwei** [open-mmlab/mmdetection#876]
Hi, I noticed that with default configuration like:
```
optimizer = dict(type='SGD', lr=0.005, momentum=0.9, weight_decay=0.0001)
optimizer_config = dict(grad_clip=dict(max_norm=35, norm_type=2))
lr_config = dict(        
    policy='step',       
    warmup='linear',     
    warmup_iters=500,    
    warmup_ratio=1.0 / 3,
    step=[16, 21])    
```
the learning rate will decrease to 1/10 when training reaches a plateau (several epochs/iters?). 
Now I want to change those settings, like in pytorch we have 'patience' or so. What is the api name for that ? Are there any documentation I can refer to? Thanks!

**Comment by leemengwei** [comment]
In `mmcv/runner/runner.py` I saw `hook_name = lr_config['policy'].title() + 'LrUpdaterHook'`
and in `mmcv/mmcv/runner/hooks/lr_updater.py` I saw things like `class FixedLrUpdaterHook(LrUpdaterHook):` or `class StepLrUpdaterHook(LrUpdaterHook)`
Thus, learning rate planner in configuration file of mmd should be like
lr_config = dict(        
    policy='step',         #Here, could alter to fixed, step, exp, poly, inv, cosine
    warmup='linear',     
    warmup_iters=500,    
    warmup_ratio=1.0 / 3,
    step=[16, 21])

**Comment by rubeea** [comment]
Can anyone suggest on how can we find the optimal learning rate via using python packages such as:
https://github.com/davidtvs/pytorch-lr-finder
Where should I implement the functionality in the framework? Do I need to create a separate lr hook for this?

Any help will be greatly appreciated
Thanks

**Comment by orbitlink** [comment]
@rubeea Have you found a solution to use [pytorch-lr-finder](https://github.com/davidtvs/pytorch-lr-finder) ? Or found the optimal learning rate ? Thanks a lot 😃

**Comment by rubeea** [comment]
> @rubeea Have you found a solution to use [pytorch-lr-finder](https://github.com/davidtvs/pytorch-lr-finder) ? Or how found the optimal learning rate ? Thanks a lot 😃

Hi,

I tried to implement pytorch-lr-finder but couldn't do so due to separability in components in mmesegmentation framework. Hence, I naively experimented with several learning rates until I found the one that helps in model convergence.

**Comment by orbitlink** [comment]
@rubeea Thanks. After reflection `poly` schedule seems to be able to help to find the optimal lr. You can fix indirectly _start_lr_ and _end_lr_ by fix _min_lr_ in `lr_config = dict(policy='poly', power=0.9, min_lr=1e-4, by_epoch=False)`, and your initial lr in your optimizer.
